### ISSUE TITLE
Prevent flatting of 3D Terrain Satellite map styles. 

### ISSUE THREAD

**Issue by Gnative** [mapbox/mapbox-maps-ios#2240]
I updated my Satellite map style and using the 3D Terrain layer to display this. Now when I pitch to 0 (or viewing directly downward) the map does the flatting animation thing where the 3D Terrain layer is flattened.. I didn't set the map to do this and would prefer it wouldn't do this as it looks a little buggy. Is this a new feature and if so can we have a setting to keep it so that the 3D perspective is kept ? 

Thank you

Strangely the vector maps still have this 3D perspective on 0 pitch, its only applied to the satellite map style as you can see in this video.

![Simulator Screenshot - iPhone 16 Pro - 2024-09-21 at 18 51 48](https://github.com/user-attachments/assets/ab96bdff-57b1-49b5-b910-fbc979a1865b)

**Comment by astojilj** [comment]
Could you please check if your style has the root-level property "camera"->"camera-projection" set to "perspective"? Flattening is related to "orthographic" "camera-projection" behavior. Standard style uses "orthographic".

E.g. style's json should not have "orthographic" here but "perspective".
 
![image](https://github.com/user-attachments/assets/e4b42c18-e45d-402f-a7be-b320e2990aa4)

**Comment by Gnative** [comment]
Hi @astojilj thank for the reply. 

No it looks like the root-level camera property is not set.. This is a fairly old base style that I have updated. 
I am not able to see where you would set this with in the style either, sorry if this is obvious. It is just not to me at the moment.

[style.json](https://github.com/user-attachments/files/17090545/style.json)

**Comment by astojilj** [comment]
@Gnative 

Your style now imports Standard Satellite style, and inherits the layers and settings from there, including camera-projection. The style.json you shared includes this:


    "imports": [
        {
            "id": "basemap",
            "url": "mapbox://styles/mapbox/standard-satellite",
            "config": {"show3dObjects": false, "theme": "default"}
        }
    ],

So, you'd need to override this - add this e.g. just before "imports", the code should look like:

    "camera": {
        "camera-projection": "orthographic"
    },
    "imports": [
        {
            "id": "basemap",
            "url": "mapbox://styles/mapbox/standard-satellite",
            "config": {"show3dObjects": false, "theme": "default"}
        }
    ],

**Comment by Gnative** [comment]
OK thank you so much. I don't see where to do this in the Style Studio. I guess I need to do this at app level ?

**Comment by astojilj** [comment]
Could you please try editing the style.json in a text editor (to add the aforementioned camera projection) and then importing the JSON file back into Studio?

**Comment by Gnative** [comment]
@astojilj  Yes this works.. the  `camera-projection:  **perspective**` property does as I expect now. I also check with the `orthographic` and this brought back the flattening effect. 

It must be when I updated a layer in the original style that this base default properties were then applied.
